examples/launch/launch-sweeps/optuna-scheduler/README.md

```diff
@@ -0,0 +1,150 @@
+
+2. Pruners
+
+Identical to the samplers above, pruners, found [here](https://optuna.readthedocs.io/en/stable/reference/pruners.html) can be instrumented. 
+
+Example: 
+
+```yaml
+# optuna_config_basic.yaml
+...
+scheduler:
+   settings:
+      pruner:
+        # https://optuna.readthedocs.io/en/stable/reference/generated/optuna.pruners.PercentilePruner.html
+        type: PercentilePruner
+        args:
+           percentile: 25.0  # kill 75% of runs
+           n_warmup_steps: 10  # pruning disabled for first x steps
+...
+```
+
+3. Pythonic search spaces
+
+Rather than defining the parameters to sweep over in yaml or json format, Optuna allows for defining the search space in a python file. For example, here is a very basic yaml hyperparameter configuration, and the subsequent python format. 
+
+```yaml
+...
+parameters:
+   epochs:
+      values: [50, 100, 200]
+   learning_rate:
+      min: 0.001
+      max: 0.1
+
+```
+
+To convert this into python, we use the Optuna `trial` object, which use to request values in a distribution from. This function is ultimately evaluated inside the OptunaScheduler job and converted into a config, which is then used to launch runs. Example python conversion to above: 
+
+```python
+
+def objective(trial):
+    epochs = trial.suggest_categorical('epochs', [50, 100, 200])
+    learning_rate = trial.suggest_float('learning_rate', 0.001, 0.1)
+
+    # because we aren't actually training in this function, just creating 
+    #    our hyperparameter space, return -1
+    return -1
+```
+
+Although more verbose, this method of constructing hyperparameter spaces comes with one huge advantage, conditional logic. For example, if we wanted to condition our batch size on the database, we might construct a pythonic search space like so: 
+
+```python
+
+def objective(trial):
+    database = trial.suggest_categorical('database', ['small', 'medium', 'large'])
+    
+    if database in ['small', 'large']:
+        batch_size = trial.suggest_categorical('batch_size', [16, 32, 64])
+
+        # maybe test randomization when the batch_size is small
+        randomize = trial.suggest_categorical('randomize', [True, False])
+    else:
+        batch_size = trial.suggest_categorical('batch_size', [64, 128, 256])
+
+    return -1
+```
+
+To use a pythonic search space in the OptunaScheduler, there are two methods. First is to create an artifact (of `type="optuna"`) with a python file containing a function exactly: `def objective(trial)` which will be loaded up and used as the sweep hyperparameter space. The python filename inside the artifact can be specified in the sweep config as: `optuna_source_filename`, or otherwise will assumed to be named: `optuna_wandb.py`. The second is to include that file in the log_code step when creating a code artifact scheduler job. Both methods require setting a filepath parameter in the sweep config, which the Optuna scheduler will use to identify which file contains the `objective` function.
```

Comment:
Lets break this paragraph out into 2 separate flows outlining how they can use the sampler. Might make sense to add this stuff to a tutorial on our docs page